Week10NetworkingCodingExercise/Week10NetworkingCodingExercise/Program.cs

```diff
@@ -0,0 +1,168 @@
+            UriBuilder uri = new UriBuilder("http", "localhost", portNumber);
+
+            Console.WriteLine($"Query: {uri.Query}");
+
+            if (uri.Query != "")
+            {
+                ListenAsync(uri);
+            }
+            else
+            {
+                // Create a listener.
+                using (HttpListener listener = new HttpListener())
+                {
+                    listener.Prefixes.Add(uri.ToString());
+                    listener.Start();
+
+                    Console.WriteLine("Listening...");
+
+                    // Note: The GetContext method blocks while waiting for a request. 
+                    HttpListenerContext context = listener.GetContext();
+                    HttpListenerRequest request = context.Request;
+
+                    // Obtain a response object.
+                    HttpListenerResponse response = context.Response;
+
+                    // Construct a response.
+                    string responseString = "<HTML><BODY> Hello world!</BODY></HTML>";
+                    byte[] buffer = System.Text.Encoding.UTF8.GetBytes(responseString);
+
+                    // Get a response stream and write the response to it.
+                    response.ContentLength64 = buffer.Length;
+                    Stream output = response.OutputStream;
+                    output.Write(buffer, 0, buffer.Length);
+
+                    // You must close the output stream.
+                    output.Close();
+                    listener.Stop();
+                }
+            }
+
+            Console.WriteLine($"Server running on port {portNumber}. Press Enter to stop.");
+            Console.ReadLine();
+        }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="url"></param>
+        async static void ListenAsync(UriBuilder url)
+        {
+            UriBuilder uri = url;
+
+            HttpListener listener = new HttpListener();
+            listener.Prefixes.Add(uri.ToString());
+            listener.Start();
+
+            while (true)
+            {
+                try
+                {
+                    var context = await listener.GetContextAsync();
+                    await Task.Run(() => ProcessRequestAsync(context));
+                }
+                catch (HttpListenerException) { break; }
+                catch (InvalidOperationException) { break; }
+            }
+
+            listener.Stop();
+        }
+
+        /// <summary>
+        /// Calculates values from the query string. 
+        /// </summary>
+        /// <param name="context"></param>
+        private static void ProcessRequestAsync(HttpListenerContext context)
+        {
+            string x = context.Request.QueryString["x"];
+            string y = context.Request.QueryString["y"];
+
+            if (!string.IsNullOrWhiteSpace(x) && !string.IsNullOrWhiteSpace(y))
+            {
+                string result = x + y;
```

Comment:
For the assignment, the provided numbers should be added together rather than using string concatenation to combine the two numbers. For this to work, you need to try to convert the querystring values to integers so they can be added together. You can use int.TryParse for this.